x-pack/plugin/transform/src/main/java/org/elasticsearch/xpack/transform/utils/SearchExceptionRootCauseFinder.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+package org.elasticsearch.xpack.transform.utils;
+
+import org.elasticsearch.ElasticsearchException;
+import org.elasticsearch.action.search.SearchPhaseExecutionException;
+import org.elasticsearch.action.search.ShardSearchFailure;
+
+/**
+ * Set of static utils to find the cause of a search exception.
+ */
+public final class SearchExceptionRootCauseFinder {
+
+    /**
+     * Unwrap the Exceptionstack and return the most likely cause.
+     *
+     * @param e raw Exception
+     * @return unwrapped elasticsearch exception
+     */
+    public static ElasticsearchException getRootCauseElasticsearchException(Exception e) {
```

Comment:
Why didn't you put this in the transforms ExceptionHelper class?

Reply:
ExceptionHelper is used to validate input parameters, so it serves a different purpose. In the long run this generic class should move into core. 

The root cause finder has a different purpose and probably can not be generalized.